### PR TITLE
[DA-3702](dbt) Add geo info to enriched tables

### PR DESCRIPTION
**Pull request by MatthieuRipollPassCulture**
# DE PR

## Describe your changes

This PR is linked to [this ticket](https://www.notion.so/passcultureapp/Ajouter-les-infos-territorialis-es-dans-les-tables-enrichies-4bfa16cad25c4dda8d5d706692fd9e97).

The main goal is to delete the table`analytics_prod.user_locations` and to add the geo infos into the enriched models.

Tag a reviewer if necessacy  @github/username 

## Jira ticket number and/or notion link

JIRA-ticket_number

### Type of change
- [ ] Fix (non-breaking change which corrects expected behavior)
- [ ] New fields (non-breaking change)
- [ ] New table (non-breaking change)
- [ ] Concept change (potentially breaking change which modifies fields according to new or evolving business concepts) 
- [ ] Table deletion (potentially breaking change which adds functionality/ table)
      
### Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] My code passes CI/CD tests
- [ ] I updated README.md
- [ ] I have updated the dag
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [ ] I will create a review on slack and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### Added tests?
- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] ⏰ no, but I created a ticket

### CODE REVIEW

File: orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql [hunk 1]
```diff
@@ -35,11 +41,16 @@ SELECT
     collective_booking.educational_year_id AS educational_year_id,
     educational_year.scholar_year,
     collective_booking.educational_redactor_id AS educational_redactor_id,
-    eple.nom_etablissement,
+    educational_institution.institution_name AS nom_etablissement,
     institution_program.institution_program_name AS institution_program_name,
-    eple.code_departement AS school_department_code,
-    school_region_departement.region_name AS school_region_name,
-    eple.libelle_academie,
+    institution_locations.institution_academy AS libelle_academie,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql:46]:
```suggestion
    institution_locations.institution_academy AS institution_academy_name,
```

File: orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql [hunk 2]
```diff
@@ -35,11 +41,16 @@ SELECT
     collective_booking.educational_year_id AS educational_year_id,
     educational_year.scholar_year,
     collective_booking.educational_redactor_id AS educational_redactor_id,
-    eple.nom_etablissement,
+    educational_institution.institution_name AS nom_etablissement,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql:44]:
```suggestion
    educational_institution.institution_name AS institution_name,
```

File: orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql [hunk 3]
```diff
@@ -35,11 +41,16 @@ SELECT
...
     institution_program.institution_program_name AS institution_program_name,
-    eple.code_departement AS school_department_code,
-    school_region_departement.region_name AS school_region_name,
-    eple.libelle_academie,
+    institution_locations.institution_academy AS libelle_academie,
+    institution_locations.institution_region_name AS school_region_name,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql:47]:
```suggestion
    institution_locations.institution_region_name AS institution_region_name,
```

File: orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql [hunk 4]
```diff
@@ -35,11 +41,16 @@ SELECT
...
+    institution_locations.institution_department_code AS school_department_code,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql:48]:
```suggestion
    institution_locations.institution_department_code AS institution_department_code,
```

File: orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_institution_data.sql [hunk 5]
```diff
@@ -196,14 +196,16 @@ SELECT
     educational_institution.institution_type,
     eple_aggregated_type.macro_institution_type,
     institution_program.institution_program_name AS institution_program_name,
-    region_department.academy_name AS institution_academie,
-    region_department.region_name AS institution_region_name,
+    location_info.institution_academy,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_institution_data.sql:199]:
```suggestion
    location_info.institution_academy_name,
```

File: orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_institution_data.sql [hunk 6]
```diff
@@ -196,14 +196,16 @@ SELECT
...
-    educational_institution.institution_city,
-    rurality.geo_type as institution_rural_level,
-    location_info.institution_latitude as institution_latitude,
-    location_info.institution_longitude as institution_longitude,
+    location_info.institution_city,
+    location_info.institution_epci,
+    location_info.institution_density_label,
+    location_info.institution_macro_density_label, 
+    location_info.institution_latitude,
+    location_info.institution_longitude,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_institution_data.sql:208]:
```suggestion
    location_info.institution_longitude,
    CASE WHEN location_info.institution_qpv_name is not null THEN TRUE ELSE FALSE END AS institution_in_qpv,
```

File: orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_user_data.sql [hunk 7]
```diff
@@ -455,7 +454,13 @@ SELECT
     user.user_has_enabled_marketing_email,
     user.user_iris_internal_id,
     themes_subscribed.currently_subscribed_themes,
-    themes_subscribed.is_theme_subscribed
+    themes_subscribed.is_theme_subscribed,
+    user_region_name,
+    user.user_city,
+    user.user_epci,
+    user.user_academy,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_user_data.sql:461]:
```suggestion
    user.user_academy_name,
```

File: orchestration/dags/data_gcp_dbt/models/analytics/institution_locations.sql [hunk 8]
```diff
@@ -48,6 +46,12 @@ institutions_qpv AS (
 )
 SELECT
   iq.*,
+  gi.city_label AS institution_city,
+  gi.region_name AS institution_region_name,
+  gi.epci_label AS institution_epci,
+  gi.academy_name AS institution_academy,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/analytics/institution_locations.sql:52]:
```suggestion
  gi.academy_name AS institution_academy_name,
```

File: orchestration/dags/data_gcp_dbt/models/clean/user_beneficiary.sql [hunk 9]
```diff
@@ -101,6 +101,12 @@ SELECT
         ELSE user_creation_date
     END AS user_activation_date,
     ui.user_iris_internal_id,
+    ui.user_region_name,
+    ui.user_city,
+    ui.user_epci,
+    ui.user_academy,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/clean/user_beneficiary.sql:107]:
```suggestion
    ui.user_academy_name,
```

File: orchestration/dags/data_gcp_dbt/models/intermediate/api_gouv/address/int_api_gouv__address_user_geo_iris.sql [hunk 10]
```diff
@@ -35,6 +31,11 @@ users_with_geo_candidates AS (
     u.*,
     gi.iris_internal_id AS user_iris_internal_id,
     gi.region_name AS user_region_name,
+    gi.city_label AS user_city,
+    gi.epci_label AS user_epci,
+    gi.academy_name AS user_academy,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/intermediate/api_gouv/address/int_api_gouv__address_user_geo_iris.sql:36]:
```suggestion
    gi.academy_name AS user_academy_name,
```

File: orchestration/dags/data_gcp_dbt/models/mart/global/mrt_global__booking.sql [hunk 11]
```diff
@@ -33,7 +40,14 @@ SELECT
     s.digital_goods,
     s.event,
     s.offer_category_id,
+    u.user_postal_code,
     u.user_department_code,
+    u.user_region_name,
+    u.user_city,
+    u.user_epci,
+    u.user_academy,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/mart/global/mrt_global__booking.sql:48]:
```suggestion
    u.user_academy_name,
```

File: orchestration/dags/data_gcp_dbt/models/mart/global/mrt_global__deposit.sql [hunk 12]
```diff
@@ -22,10 +16,16 @@ SELECT
...
-    bgd.user_civility,
-    bgd.user_department_code,
-    bgd.user_age,
-    region_department.region_name AS user_region_name,
+    u.user_civility,
+    u.user_region_name,
+    u.user_postal_code,
+    u.user_city,
+    u.user_epci,
+    u.user_academy,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/mart/global/mrt_global__deposit.sql:24]:
```suggestion
    u.user_academy_name,
```

File: orchestration/dags/data_gcp_dbt/models/mart/global/mrt_global__user.sql [hunk 13]
```diff
@@ -5,6 +5,12 @@ SELECT
     user_has_enabled_marketing_email,
     user_department_code,
     user_postal_code,
+    user_region_name,
+    user_city,
+    user_epci,
+    user_academy,
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/mart/global/mrt_global__user.sql:11]:
```suggestion
    user_academy_name,
```

File: orchestration/dags/data_gcp_dbt/models/sources.yml [hunk 14]
```diff
@@ -162,7 +162,7 @@ sources:
       - name: app_native_logs
       - name: epci
       - name: eple
-      - name: institution_locations
+      - name: firebase_pro_events
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/sources.yml:165]: c'est normal ça ?

File: orchestration/dags/data_gcp_dbt/models/intermediate/api_gouv/address/int_api_gouv__address_user_location.sql [hunk 15]
```diff
@@ -7,15 +7,15 @@ SELECT
...
-    epci_name,
-    qpv_communes,
     qpv_name,
     code_qpv,
     zrr,
     date_updated,
     user_iris_internal_id,
     user_region_name,
+    user_epci,
+    user_academy,
```
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/intermediate/api_gouv/address/int_api_gouv__address_user_location.sql:17]: En tournant en dev j'ai cette erreur `Unrecognized name: user_academy at [24:5]`

File: orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__user.sql [hunk 16]
```diff
@@ -42,6 +42,11 @@ SELECT
     u.user_cultural_survey_filled_date,
     CASE WHEN u.user_role IN ("UNDERAGE_BENEFICIARY", "BENEFICIARY") THEN 1 ELSE 0 END AS is_beneficiary,
     ui.user_iris_internal_id,
-    ui.user_region_name
+    ui.user_region_name,
+    ui.user_city,
+    ui.user_epci,
+    ui.user_academy,
```
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/intermediate/applicative/int_applicative__user.sql:48]: Idem ici le nom de colonne n'est pas le bon

File: orchestration/dags/data_gcp_dbt/models/mart/global/mrt_global__booking.sql [hunk 17]
```diff
@@ -52,6 +66,8 @@ SELECT
 FROM {{ ref('int_applicative__booking') }} AS b
 INNER JOIN {{ ref('mrt_global__stock') }} AS s ON s.stock_id = b.stock_id
 INNER JOIN {{ ref('mrt_global__user') }} AS u ON u.user_id = b.user_id
+INNER JOIN {{ ref('mrt_global__venue_unverified') }} AS v ON v.venue_id = v.venue_id
```
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/mart/global/mrt_global__booking.sql:69]: Ici pas besoin d'ajouter cette jointure car les champs correspondant sont déjà dans `mrt_global__stock` de venue sont déjà dans `mrt_global__stock` :)  
-> on peut supprimer la jointure avec mrt_global__venue_unverified et remplacer les 'v.' par 's.'

File: orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql [hunk 18]
```diff
@@ -35,11 +41,16 @@ SELECT
     collective_booking.educational_year_id AS educational_year_id,
     educational_year.scholar_year,
     collective_booking.educational_redactor_id AS educational_redactor_id,
-    eple.nom_etablissement,
```
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql:38]: Question : tu as supprimé la colonne ici et non dans `mrt_global__collective_booking` : on veut pas la supprimer des deux modèles ?

File: orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql [hunk 19]
```diff
@@ -35,11 +41,16 @@ SELECT
...
+    educational_institution.institution_name AS institution_name,
     institution_program.institution_program_name AS institution_program_name,
-    eple.code_departement AS school_department_code,
-    school_region_departement.region_name AS school_region_name,
-    eple.libelle_academie,
```
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql:42]: Idem pour ces trois colonnes

File: orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql [hunk 20]
```diff
@@ -58,13 +69,10 @@ FROM
...
-    INNER JOIN {{ ref('venue') }} AS venue ON collective_booking.venue_id = venue.venue_id
+    INNER JOIN {{ ref('int_applicative__venue') }} AS venue ON collective_booking.venue_id = venue.venue_id
     INNER JOIN {{ source('raw', 'applicative_database_offerer') }} AS offerer ON offerer.offerer_id = venue.venue_managing_offerer_id
     INNER JOIN {{ ref('educational_institution') }} AS educational_institution ON educational_institution.educational_institution_id = collective_booking.educational_institution_id
     INNER JOIN {{ source('raw', 'applicative_database_educational_year') }} AS educational_year ON educational_year.adage_id = collective_booking.educational_year_id
-    LEFT JOIN {{ source('analytics', 'eple') }} AS eple ON eple.id_etablissement = educational_institution.institution_id
-    LEFT JOIN {{ source('analytics', 'region_department') }} AS venue_region_departement ON venue.venue_department_code = venue_region_departement.num_dep
-    LEFT JOIN {{ source('analytics', 'region_department') }} AS school_region_departement ON eple.code_departement = school_region_departement.num_dep
```
Comment (carolinemestre-passculture) [inline orchestration/dags/data_gcp_dbt/models/analytics/enriched/enriched_collective_booking_data.sql:67]: + du coup on pourrait supprimer ces 3 jointures dans les nouveaux dossiers aussi

**Review by carolinemestre-passculture — APPROVED** [review]

**Pull request merged**